src/context.rs

```diff
@@ -8,9 +8,18 @@ use std::{ffi, io};
 macro_rules! variadic {
     ($name:ident, $op:ident) => {
         pub fn $name<I: IntoIterator<Item = SExpr>>(&self, items: I) -> SExpr {
-            let args: Vec<_> = std::iter::once(self.atoms.$op).chain(items).collect();
-            assert!(args.len() >= 3);
-            self.list(args)
+            let mut iter = items.into_iter().peekable();
+            let first = iter.next().expect("At least one argument is expected");
```

Comment:
FWIW, `all([]) == true` and `any([]) == false`. Not sure whether we want to actually implement that however...

Reply:
That's a great point. We could add another case to `variadic!` to handle empty inputs when there's a known reasonable value.

Reply:
@avanhatt would you find ^ useful?

Reply:
Hm, I have less of a strong opinion on the empty case! It would save me a bit of special casing, but as long as I don't need separate special cases for 0 vs 1 vs. many elements, I could go either way.